course_flow/context_processors.py

```diff
@@ -5,13 +5,17 @@
 
 
 def update_notifications(request):
-    updates = UpdateNotification.objects.order_by("-created_on")
-    if updates.count() > 0:
-        last_update = updates.first()
-        last_update_serialized = (
-            JSONRenderer()
-            .render(UpdateNotificationSerializer(last_update).data)
-            .decode("utf-8")
-        )
-        return {"update_notifications": last_update_serialized}
+    try:
+        if "course-flow" in request.path:
+            updates = UpdateNotification.objects.order_by("-created_on")
+            if updates.count() > 0:
```

Comment:
I think the `.count()` here is more expensive than a simple `.exists()` for the if statement (my tests with dalite Question model is about a factor of 2).   I'd go with that and just make the full query in the if statement.

Reply:
You know, I don't think I've ever realized .exists() is a thing. I've probably got a lot of count()s that don't need to be in there. I'll fix this one for now and maybe create a jira issue to remind myself to go back and hunt down more later.